### PR TITLE
[GCS]Add gcs dump log(Part1)

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
Add gcs dump log. We will get the session_dir and write the log to gcs_debug_state.txt in the next PR.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc [hunk 1]
```diff
@@ -1132,5 +1142,27 @@ void GcsActorManager::AddDestroyedActorToCache(const std::shared_ptr<GcsActor> &
       actor->GetActorID(), (int64_t)actor->GetActorTableData().timestamp());
 }
 
+std::string GcsActorManager::DumpDebugMetrics() const {
```
Comment (kfstorm) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:1145]: Rename to `DebugString`

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc [hunk 2]
```diff
@@ -1132,5 +1142,27 @@ void GcsActorManager::AddDestroyedActorToCache(const std::shared_ptr<GcsActor> &
...
+         << "\nUpdateActorInfo count: " << metrics_[MetricsType::UPDATE_ACTOR_INFO]
+         << "\nAddActorCheckpoint: " << metrics_[MetricsType::ADD_ACTOR_CHECKPOINT]
+         << "\nGetActorCheckpoint: " << metrics_[MetricsType::GET_ACTOR_CHECKPOINT]
+         << "\nGetActorCheckpointID: " << metrics_[MetricsType::GET_ACTOR_CHECKPOINT_ID]
+         << "\nRegistered actors count: " << registered_actors_.size()
+         << "\nDestroyed actors count: " << destroyed_actors_.size()
+         << "\nNamed actors count: " << named_actors_.size()
+         << "\nUnresolved actors count: " << unresolved_actors_.size()
+         << "\nPending actors count: " << pending_actors_.size()
+         << "\nCreated actors count: " << created_actors_.size() << "\n}";
```
Comment (kfstorm) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:1163]: Having newline characters in a debug string is probably a bad idea. Please refer to other existing implementations of `DebugString`s. Please also make sure to use a consistent format.

**Review by kfstorm — COMMENTED** [review]
You are trying to add statistics for requests received by GCS, right? Some suggestions:

1. Replace "metrics" with "stats". I think the term "metrics" has a special meaning and usually there are other components to collect and visualize metrics.

2. Rename `MetricsType::FOO_BAR` to `MetricsType::FOO_BAR_REQUEST`. And mention the "request" concept in the output string. Because at first, I don't know they are about the count of received requests. When I saw `CreateActor count`, I thought of the amount of created actors.

**Comment by rkooo567** [comment]
I prefer counts instead of stats because they are used to figure out if there are leaks. 

As @kfstorm We will use a term metrics for actual ray metrics (metrics_def.h)

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc [hunk 3]
```diff
@@ -120,6 +120,7 @@ void GcsActorManager::HandleRegisterActor(const rpc::RegisterActorRequest &reque
                    << ", job id = " << actor_id.JobId() << ", actor id = " << actor_id;
     GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
   }
+  ++counts_[CountsType::REGISTER_ACTOR_REQUEST];
```
Comment (kfstorm) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:123]:
```suggestion
  ++counts_[CountType::REGISTER_ACTOR_REQUEST];
```

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc [hunk 4]
```diff
@@ -1137,5 +1147,33 @@ void GcsActorManager::AddDestroyedActorToCache(const std::shared_ptr<GcsActor> &
       actor->GetActorID(), (int64_t)actor->GetActorTableData().timestamp());
 }
 
+std::string GcsActorManager::DebugString() const {
+  std::ostringstream stream;
+  stream << "GcsActorManager: {RegisterActor request counts: "
```
Comment (kfstorm) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:1152]:
```suggestion
  stream << "GcsActorManager: {RegisterActor request count: "
```

**Review by kfstorm — APPROVED** [review]

**Comment by kfstorm** [comment]
@rkooo567 any more comments?

**Comment by rkooo567** [comment]
Nope! Feel free to merge it!

**Pull request merged**